x/posts/client/cli/tx.go

```diff
@@ -67,11 +79,38 @@ func GetCmdCreatePost(cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
-			msg := types.NewMsgCreatePost(args[1], parentID, allowsComments, args[0], map[string]string{}, from, time.Now().UTC())
+			var msg types.MsgCreatePost
+			msg = types.NewMsgCreateTextPost(
+				args[1],
+				parentID,
+				allowsComments,
+				args[0],
+				map[string]string{},
+				from,
+				time.Now().UTC(),
+			)
+
+			// If there are some medias
+			if len(args) > 3 {
+				var medias types.PostMedias
+				// Read each media and add it to the medias if valid
+				for i := 3; i < len(args); i++ {
+					arg := strings.Split(args[i], ",")
+					if len(arg) == 3 {
+						media := types.NewPostMedia(arg[0], arg[1], arg[2])
+						medias = append(medias, media)
+					} else {
+						return sdk.ErrUnknownRequest("If medias are present, you should specify uri, provider and mime type, if you are confused, please use the --help flag")
+					}
+				}
+				if textMsg, ok := msg.(types.MsgCreateTextPost); ok {
```

Comment:
This check will always succeed, can we simply delete it and replace it with something like the following?

```golang
msg = types.NewMsgCreateMediaPost(MsgCreateTextPost(msg), medias))
```